plugins/outputs/postgresql/utils/utils.go

```diff
@@ -0,0 +1,158 @@
+// [m, m, m2, m2, m] => {m:[0,1,4], m2:[2,3]}
+func GroupMetricsByMeasurement(m []telegraf.Metric) map[string][]int {
+	toReturn := make(map[string][]int)
+	for i, metric := range m {
+		var metricLocations []int
+		var ok bool
+		name := metric.Name()
+		if metricLocations, ok = toReturn[name]; !ok {
+			metricLocations = []int{}
+			toReturn[name] = metricLocations
+		}
+		toReturn[name] = append(metricLocations, i)
+	}
+	return toReturn
+}
+
+// BuildJsonb returns a byte array of the json representation
+// of the passed object.
+func BuildJsonb(data interface{}) ([]byte, error) {
+	d, err := json.Marshal(data)
+	if err != nil {
+		return nil, err
+	}
+	return d, nil
+}
+
+// QuoteIdent returns a sanitized string safe to use in SQL as an identifier
+func QuoteIdent(name string) string {
+	return pgx.Identifier{name}.Sanitize()
+}
+
+// QuoteLiteral returns a sanitized string safe to use in sql as a string literal
+func QuoteLiteral(name string) string {
+	return "'" + strings.Replace(name, "'", "''", -1) + "'"
+}
+
+// FullTableName returns a sanitized table name with it's schema (if supplied)
+func FullTableName(schema, name string) *pgx.Identifier {
+	if schema != "" {
+		return &pgx.Identifier{schema, name}
+	}
+
+	return &pgx.Identifier{name}
+}
+
+// DerivePgDatatype returns the appropriate PostgreSQL data type
+// that could hold the value.
+func DerivePgDatatype(value interface{}) PgDataType {
+	switch value.(type) {
+	case bool:
+		return "boolean"
+	case uint64:
+		return "int8"
+	case int64:
+		return "int8"
+	case int:
+		return "int8"
+	case uint:
+		return "int8"
+	case uint32:
+		return "int4"
+	case int32:
+		return "int4"
+	case float64:
+		return "float8"
+	case float32:
+		return "float8"
+	case string:
+		return "text"
+	case time.Time:
+		return "timestamptz"
+	default:
+		log.Printf("E! Unknown datatype %T(%v)", value, value)
+		return "text"
+	}
+}
+
+// LongToShortPgType returns a PostgreSQL datatype in it's short
+// notation form.
+func LongToShortPgType(longPgType string) PgDataType {
+	switch longPgType {
+	case "boolean":
+		return "boolean"
+	case "integer":
+		return "int4"
+	case "bigint":
+		return "int8"
+	case "double precision":
+		return "float8"
+	case "text":
+		return "text"
+	case "timestamp with time zone":
+		return "timestamptz"
+	case "timestamp without time zone":
+		return "timestamp"
+	case "jsonb":
+		return "jsonb"
+	default:
+		log.Printf("E! Unknown datatype %s", longPgType)
+		return PgDataType(longPgType)
+	}
+}
+
+// PgTypeCanContain tells you if one PostgreSQL data type can contain
+// the values of another without data loss.
+func PgTypeCanContain(canThis PgDataType, containThis PgDataType) bool {
+	if canThis == containThis {
+		return true
+	}
+	if canThis == "int8" {
+		return containThis == "int4"
+	}
+	if canThis == "int4" {
+		return containThis == "serial"
+	}
+	if canThis == "float8" {
+		return containThis == "int4" || containThis == "int8"
```

Comment:
can `float8` really contain `int8` in PG? `float8` can only represent a subset of integers in `int8` so I’m surprised they let you do that conversion

Reply:
My bad, it can't be stored only up to 2^53